File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        List<Integer> temp = new ArrayList<>();
        for(int i = 0; i < score.length; i++){
            temp.add(score[i]);
            Collections.sort(temp); 
            if(temp.size() <= k) answer[i] = temp.get(0);
            else answer[i] = temp.get(temp.size() - k);

        }
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    let answer = [];
    let temp = [];
    for(let i = 0; i < score.length; i++){
        temp.push(score[i]);
        temp.sort((a, b) => a - b);
        if(temp.length <= k) answer.push(temp[0]);
        else answer.push(temp[temp.length - k]);
        
    }
    return answer;
}